tests/test_hpack.c

```diff
@@ -701,3 +701,104 @@ static int test_hpack_dynamic_table_get(struct aws_allocator *allocator, void *c
     aws_http_library_clean_up();
     return AWS_OP_SUCCESS;
 }
+
+static int s_check_header(
+    const struct aws_http_header *header_field,
+    const char *name,
+    const char *value,
+    enum aws_http_header_compression compression) {
+
+    ASSERT_BIN_ARRAYS_EQUALS(name, strlen(name), header_field->name.ptr, header_field->name.len);
+    ASSERT_BIN_ARRAYS_EQUALS(value, strlen(value), header_field->value.ptr, header_field->value.len);
+    ASSERT_INT_EQUALS(compression, header_field->compression);
+    return AWS_OP_SUCCESS;
+}
+
+AWS_TEST_CASE(hpack_decode_indexed_from_dynamic_table, test_hpack_decode_indexed_from_dynamic_table)
+static int test_hpack_decode_indexed_from_dynamic_table(struct aws_allocator *allocator, void *ctx) {
+    (void)ctx;
+
+    aws_http_library_init(allocator);
+    struct aws_hpack_context *context = aws_hpack_context_new(allocator, AWS_LS_HTTP_GENERAL, NULL);
+    ASSERT_NOT_NULL(context);
+
+    /* clang-format off */
+    uint8_t input[] = {
+        0x48, 0x03, '3', '0', '2',      /* ":status: 302" - stored to dynamic table */
+        0x40, 0x01, 'a', 0x01, 'b',     /* "a: b" - stored to dynamic table */
```

Comment:
Just FYI: there's another file `test_h2_headers.c` that has a lot of tests that are much more complex than what existed here in `test_hpack.c`. It's actually composed of the samples from [RFC-7541 Section C](https://httpwg.org/specs/rfc7541.html#rfc.section.C.3.1). The problem is, we don't have ALL the samples. If we did we would have picked up this bug already. Looks like we stopped short including any examples that spanned multiple messages, so we didn't catch this bug where we insert something in the table and look it up again later.

In the future we should strongly consider getting the rest of the examples into that other test file